### DISCUSSION TITLE
Is there a way to render these dynamic divs with the same width of the first four?

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#12769] [Help]
As you can see in this Tailwind REPL: https://play.tailwindcss.com/in8MDw0Zwb

the 5, 6 and 7 div are bigger than the 1, 2, 3 and 4.

Is there a way to render them with the same width of the first four and centering them?

I cannot control how many divs are there on each render. All is dynamic.

[![enter image description here][1]][1]

    <div class="bg-gray-300 p-20">
      <div class="flex flex-wrap gap-6 bg-red-400">
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">1</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">2</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">3</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">4</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">5</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">6</div>
        <div class="flex-shrink-0 flex-grow basis-1/5 bg-white text-center">7</div>
      </div>
    </div>

  [1]: https://i.stack.imgur.com/62CIk.png

**Comment by wongjn** [comment]
I don't believe there is a maintainable CSS-only solution ([here's the most maintainable I could come up with right now, it is limited in its emulated behavior](https://play.tailwindcss.com/FpxHA9IIC4)), but it can be done with JavaScript. [See this CodePen for an example](https://codepen.io/Wongjn/pen/XWGMJGa).

**Reply by frederikhors** [reply]
Wow. Thanks. I cannot believe this is not possible today without JS.

And what if I use `grid`?

**Reply by wongjn** [reply]
Probably even worse with grid.

**Reply by frederikhors** [reply]
Wow. I'm speechless.

**Comment by olets** [comment]
One way to do this is with colored borders on the cells, negative margins on the grid, and an overflow:hidden wrapper

https://play.tailwindcss.com/HIpIbwEAic

**Comment by ony3000** [comment]
https://play.tailwindcss.com/0xV9BEOH5B

If you specify the div as fixed width, adding `justify-center` is sufficient.

**Reply by frederikhors** [reply]
Can I ask you to better explain what is goiing on in this amazing answer?

**Reply by ony3000** [reply]
@frederikhors Flex items are spaced at 24px by the `gap-6` class, so the sum of the widths taken up by the four divs is `100%-72px`, which is the total width minus three gaps.

Therefore, the width occupied by one div is calculated as `(100%-72px)/4 = 25%-18px`.

(Strictly speaking, it is not a "fixed width" because it includes a percentage. :joy:)
